pkg/provider/aws/expose.go

```diff
@@ -0,0 +1,123 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/klothoplatform/klotho/pkg/annotation"
+	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/klothoplatform/klotho/pkg/multierr"
+	"github.com/klothoplatform/klotho/pkg/provider/aws/resources"
+	"github.com/pkg/errors"
+)
+
+func (a *AWS) GenerateExposeResources(gateway *core.Gateway, result *core.ConstructGraph, dag *core.ResourceGraph) error {
+	err := a.CreateRestApi(gateway, result, dag)
```

Comment:
Is this just a "self-documenting code" thing, or is there a reason for this indirection? Ie, why not just have the `CreateRestApi` body be this body"

Reply:
i mainly just split it out because when we add other resources that can get created from expose, it would be an ugly function